### PR TITLE
apple_sdk: 10.12 -> 10.13

### LINKED ISSUE #101229: Update macOS SDK
I'm seeing a growing number of failures on hydra for packages requiring the 10.13 SDK on macOS. What is the timescale on a possible update, and what would that involve? Alternatively, would there be some way to build certain packages against the 10.13 SDK while keeping the default stdenv and majority of packages 10.12-compatible?

Related:
https://github.com/NixOS/nixpkgs/pull/99133 (folly)
https://github.com/NixOS/nixpkgs/pull/101125 (apk-tools)
https://github.com/NixOS/nixpkgs/issues/97479 (various)

### PR DESCRIPTION
**Pull request by SuperSandro2000**
###### Motivation for this change

Closes #101229

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
I changed the target branch to staging because we are rebuilding basically everything.

File: pkgs/os-specific/darwin/apple-sdk/default.nix [hunk 1]
```diff
@@ -11,14 +11,14 @@ let
     pname = "MacOS_SDK";
     inherit version;
 
-    # This URL comes from https://swscan.apple.com/content/catalogs/others/index-10.12.merged-1.sucatalog, which we found by:
-    #  1. Google: site:swscan.apple.com and look for a name that seems appropriate for your version
+    # This URL comes from https://swscan.apple.com/content/catalogs/others/index-10.13-1.sucatalog.gz, which we found by:
+    #  1. DuckDuckGo: site:swscan.apple.com and DevSDK .pkg
```
Comment (r-burns) [inline pkgs/os-specific/darwin/apple-sdk/default.nix:15]: 😆 👍

Comment (SuperSandro2000) [reply pkgs/os-specific/darwin/apple-sdk/default.nix:15]: Google gave me 1 result for the domain and DuckDuckGo has plenty to work with.

**Review by LnL7 — COMMENTED** [review]
This doesn't include any apple source releases which are most likely also needed and the tricky part of bumping the sdk version.

**Comment by r-burns** [comment]
@LnL7 What's tricky about the apple source releases? Is there something non-experts (like me) can do to help out with the update?

**Comment by LnL7** [comment]
Hard to say anything specific. Just in general these components are often only partially opensourced and the fact that we don't use xcode or whatever apple uses internally makes things usually a bit more complicated compared to projects with an actual opensource community.

For reference this where the changes for 10.12 https://github.com/NixOS/nixpkgs/pull/56744 (https://github.com/NixOS/nixpkgs/commit/e6f7f2928f668262577125290dc90ad8df12e724).

Updating [CF](https://github.com/NixOS/nixpkgs/blob/d3badf3b435f423ed666a694af4cf1f301a898a9/pkgs/os-specific/darwin/swift-corelibs/corefoundation.nix) is probably one of the main things that's missing, along with some other core parts like Libc, Libsystem, etc. Those will probably drive most of the other changes needed based on what problems arise from updating those.

**Comment by siraben** [comment]
Any updates on this PR?

**Comment by SuperSandro2000** [comment]
> Any updates on this PR?

I currently do not have more plans to work on this right now. If anyone is interested feel free to start from here.

**Pull request closed without merging**